### PR TITLE
Add new python solutions to problems 1 through 5

### PR DESCRIPTION
**Pull request by vitsalis**

### CODE REVIEW

File: prob-1/prob1.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+sum = 0;
+
+for i in range (1000):
+    if (i % 3 == 0 or i % 5 == 0):
```
Comment (dionyziz) [inline prob-1/prob1.py]: no need for `()`

File: prob-1/prob1.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+sum = 0;
+
+for i in range (1000):
```
Comment (dionyziz) [inline prob-1/prob1.py]: no space before `(`

File: prob-1/prob1.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+sum = 0;
```
Comment (dionyziz) [inline prob-1/prob1.py]: no need for `;`

File: prob-2/prob2.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
+sum = 0;
+x = 1;
+y = 2;
+
+while (y < 4 * 10**6):
+    if (y % 2 == 0):
```
Comment (dionyziz) [inline prob-2/prob2.py]: no `()`

File: prob-2/prob2.py [hunk 5]
```diff
@@ -0,0 +1,12 @@
+sum = 0;
+x = 1;
+y = 2;
+
+while (y < 4 * 10**6):
```
Comment (dionyziz) [inline prob-2/prob2.py]: no `()`

File: prob-2/prob2.py [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+    if (y % 2 == 0):
+        sum += y;
+    tmp = x + y;
+    x = y;
+    y = tmp;
```
Comment (dionyziz) [inline prob-2/prob2.py]: you don't need a tmp var, just swap in place:

```
x, y = y, x + y
```

File: prob-2/prob2.py [hunk 7]
```diff
@@ -0,0 +1,12 @@
+sum = 0;
+x = 1;
+y = 2;
```
Comment (dionyziz) [inline prob-2/prob2.py]: no need for `;`

File: prob-3/prob3.py [hunk 8]
```diff
@@ -0,0 +1,13 @@
+max = 1;
+num = 600851475143;
+
+for i in range (3,10**7):
+    if (num % i == 0):
+        prime = True;
+        for j in range (2,i - 1):
```
Comment (dionyziz) [inline prob-3/prob3.py]: space after `,`

File: prob-4/prob4.py [hunk 9]
```diff
@@ -0,0 +1,20 @@
+def palindrome(num):
+    "Checks if the given number is a palindrome"
```
Comment (dionyziz) [inline prob-4/prob4.py]: Use docstring

File: prob-5/prob5.py [hunk 10]
```diff
@@ -0,0 +1,10 @@
+i = 20
+while (True):
+    found = True
+    for j in range(1,20):
+        if (i % j != 0):
+            found = False
```
Comment (dionyziz) [inline prob-5/prob5.py]: break?

File: prob-4/prob4.py [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+    string = str(num)
+    list = []
+    for digit in string:
+        list.append(int(digit))
```
Comment (dionyziz) [inline prob-4/prob4.py]: Why do you need to convert to int?

File: prob-4/prob4.py [hunk 12]
```diff
@@ -0,0 +1,20 @@
+def palindrome(num):
```
Comment (dionyziz) [inline prob-4/prob4.py]: rename to `is_palindrome` to indicate boolean return value

File: prob-4/prob4.py [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+    "Checks if the given number is a palindrome"
+    string = str(num)
+    list = []
```
Comment (dionyziz) [inline prob-4/prob4.py]: You can convert a string to a list using `list(string)`

File: prob-5/prob5.py [hunk 14]
```diff
@@ -0,0 +1,10 @@
+i = 20
+while (True):
```
Comment (dionyziz) [inline prob-5/prob5.py]: no `()`

File: prob-3/prob3.py [hunk 15]
```diff
@@ -0,0 +1,13 @@
+max = 1;
+num = 600851475143;
+
+for i in range (3,10**7):
+    if (num % i == 0):
```
Comment (dionyziz) [inline prob-3/prob3.py]: no `()`

File: prob-3/prob3.py [hunk 16]
```diff
@@ -0,0 +1,13 @@
+max = 1;
+num = 600851475143;
+
+for i in range (3,10**7):
```
Comment (dionyziz) [inline prob-3/prob3.py]: space after `,`

File: prob-3/prob3.py [hunk 17]
```diff
@@ -0,0 +1,13 @@
+max = 1;
+num = 600851475143;
```
Comment (dionyziz) [inline prob-3/prob3.py]: no `;`

File: prob-3/prob3.py [hunk 18]
```diff
@@ -0,0 +1,13 @@
...
+
+for i in range (3,10**7):
+    if (num % i == 0):
+        prime = True;
+        for j in range (2,i - 1):
+            if (i % j == 0):
+                prime = False;
+        if (prime):
```
Comment (dionyziz) [inline prob-3/prob3.py]: no `()`

File: prob-1/prob1.py [hunk 19]
```diff
@@ -0,0 +1,7 @@
...
+
+for i in range (1000):
+    if (i % 3 == 0 or i % 5 == 0):
+        sum += i;
```
Comment (dionyziz) [inline prob-1/prob1.py]: This for loop is written better in functional format: First collect the items you want using a comprehension, then sum them using the `sum` function.

File: prob-4/prob4.py [hunk 20]
```diff
@@ -0,0 +1,20 @@
...
+    for digit in string:
+        list.append(int(digit))
+    isPalindrome = True
+    for i in range ((len(list) // 2)):
+        if (list[i] != list[len(list) - (i+1)]):
+            isPalindrome = False
+    return isPalindrome
```
Comment (dionyziz) [inline prob-4/prob4.py]: This whole function can be written as

```
num = str(num)
return num == num[::-1]
```

File: prob-4/prob4.py [hunk 21]
```diff
@@ -0,0 +1,20 @@
...
+
+max = 1;
+for i in range (999,900,-1):
+    for j in range (999,900,-1):
```
Comment (dionyziz) [inline prob-4/prob4.py]: no space before `(`

File: prob-6/prob6.py [hunk 22]
```diff
@@ -0,0 +1,6 @@
+s1 = s2 = 0
+
+for i in range(1, 101):
+    s1 += i ** 2
+    s2 += i
+print(s2 ** 2 - s1)
```
Comment (dionyziz) [inline prob-6/prob6.py]: s1, s2 are bad names, you can use `sum_of_squares` and `square_of_sum`

File: prob-6/prob6.py [hunk 23]
```diff
@@ -0,0 +1,6 @@
+s1 = s2 = 0
+
+for i in range(1, 101):
+    s1 += i ** 2
+    s2 += i
```
Comment (dionyziz) [inline prob-6/prob6.py]: This for loop is better written by first collecting the elements, then using `sum` function

File: prob-1/prob1.py [hunk 24]
```diff
@@ -0,0 +1,7 @@
+sum = 0
+
+for i in range(1000):
+    if i % 3 == 0 or i % 5 == 0:
+        sum += i
```
Comment (dionyziz) [inline prob-1/prob1.py]: Better collected using a comprehension, then summed using `sum`

File: prob-3/prob3.py [hunk 25]
```diff
@@ -0,0 +1,13 @@
+max = 1
+num = 600851475143
+
+for i in range(3,10**7):
+    if num % i == 0:
+        prime = True
```
Comment (dionyziz) [inline prob-3/prob3.py]: Flag better named `is_prime`

File: prob-3/prob3.py [hunk 26]
```diff
@@ -0,0 +1,13 @@
...
+        for j in range(2, i - 1):
+            if i % j == 0:
+                prime = False
```
Comment (dionyziz) [inline prob-3/prob3.py]: `break`?

File: prob-4/prob4.py [hunk 27]
```diff
@@ -0,0 +1,20 @@
...
+        list.append(int(digit))
+    isPalindrome = True
+    for i in range((len(list) // 2)):
+        if list[i] != list[len(list) - (i+1)]:
+            isPalindrome = False
+    return isPalindrome
+
+max = 1
+for i in range(999,100,-1):
+    for j in range(999,100,-1):
```
Comment (dionyziz) [inline prob-4/prob4.py]: Space after `,`

File: prob-3/prob3.py [hunk 28]
```diff
@@ -0,0 +1,13 @@
+max = 1
+num = 600851475143
+
+for i in range(3,10**7):
```
Comment (dionyziz) [inline prob-3/prob3.py]: space after `,`

File: prob-4/prob4.py [hunk 29]
```diff
@@ -0,0 +1,20 @@
+def isPalindrome(num):
+    """Checks if the given number is a palindrome"""
+    string = str(num)
+    list = []
+    for digit in string:
+        list.append(int(digit))
+    isPalindrome = True
+    for i in range((len(list) // 2)):
+        if list[i] != list[len(list) - (i+1)]:
```
Comment (dionyziz) [inline prob-4/prob4.py]: space around `+`

File: prob-1/prob1.py [hunk 30]
```diff
@@ -0,0 +1,7 @@
+nums = []
+
+for i in range(1000):
+    if i % 3 == 0 or i % 5 == 0:
+        nums.append(i)
```
Comment (dionyziz) [inline prob-1/prob1.py]: Replace for loop with comprehension

File: prob-10/prob10.py [hunk 31]
```diff
@@ -0,0 +1,16 @@
+def isPrime(num):
+    """Checks if the given number is a prime"""
+    isPrime = True
+    for i in range(2, num // 2):
```
Comment (dionyziz) [inline prob-10/prob10.py]: You need to include `num // 2` in your checks

File: prob-10/prob10.py [hunk 32]
```diff
@@ -0,0 +1,16 @@
+def isPrime(num):
```
Comment (dionyziz) [inline prob-10/prob10.py]: Use underscore instead of camel case

File: prob-10/prob10.py [hunk 33]
```diff
@@ -0,0 +1,16 @@
...
+        if num % i == 0:
+            isPrime = False
+            break
+    return isPrime
+
+primes = []
+
+for i in range(2, 2 * 10 ** 6):
+    if isPrime(i):
+        primes.append(i)
```
Comment (dionyziz) [inline prob-10/prob10.py]: Use comprehension

File: prob-2/prob2.py [hunk 34]
```diff
@@ -0,0 +1,10 @@
+sum = 0
+x = 1
+y = 2
+
+while y < 4 * 10**6:
```
Comment (dionyziz) [inline prob-2/prob2.py]: spaces around `**`

File: prob-3/prob3.py [hunk 35]
```diff
@@ -0,0 +1,14 @@
+max = 1
+num = 600851475143
+
+for i in range(3, 10**7):
```
Comment (dionyziz) [inline prob-3/prob3.py]: spaces around `**`

File: prob-3/prob3.py [hunk 36]
```diff
@@ -0,0 +1,14 @@
...
+    if num % i == 0:
+        isPrime = True
```
Comment (dionyziz) [inline prob-3/prob3.py]: underscore instead of camel

File: prob-4/prob4.py [hunk 37]
```diff
@@ -0,0 +1,13 @@
+def isPalindrome(num):
```
Comment (dionyziz) [inline prob-4/prob4.py]: underscore

File: prob-4/prob4.py [hunk 38]
```diff
@@ -0,0 +1,13 @@
...
+    """Checks if the given number is a palindrome"""
+    num = str(num)
+    return num == num[::-1]
+
+max = 1
+for i in range(999, 100,-1):
+    for j in range(999, 100,-1):
```
Comment (dionyziz) [inline prob-4/prob4.py]: spaces after `,`

File: prob-5/prob5.py [hunk 39]
```diff
@@ -0,0 +1,11 @@
+i = 20
+while True:
+    found = True
+    for j in range(1,20):
+        if i % j != 0:
+            found = False
+            break
+    if found:
+        print i
+        break
```
Comment (dionyziz) [inline prob-5/prob5.py:8]: You can use idiomatic for...else and avoid having a flag if you want.

File: prob-6/prob6.py [hunk 40]
```diff
@@ -0,0 +1,7 @@
+squares = []
+nums = []
+
+for i in range(1, 101):
+    squares.append(i ** 2)
+    nums.append(i)
```
Comment (dionyziz) [inline prob-6/prob6.py]: Comprehension

File: prob-7/prob7.py [hunk 41]
```diff
@@ -0,0 +1,17 @@
+def isPrime(num):
```
Comment (dionyziz) [inline prob-7/prob7.py]: underscore

File: prob-9/prob9.py [hunk 42]
```diff
@@ -0,0 +1,12 @@
...
+for i in range(1, 1000):
+    found = False
+    for j in range(1, 1000):
+        k = math.sqrt(i ** 2 + j ** 2)
+        if i + j + k == 1000:
+            print i * j * k
+            found = True
+            break
+    if found:
+        break
```
Comment (dionyziz) [inline prob-9/prob9.py:12]: Use comprehension

File: prob-3/prob3.py [hunk 43]
```diff
@@ -0,0 +1,14 @@
...
+        for j in range(2, i - 1):
```
Comment (dionyziz) [inline prob-3/prob3.py]: Optimize with `i // 2 + 1`?

File: prob-3/prob3.py [hunk 44]
```diff
@@ -0,0 +1,14 @@
...
+            if i % j == 0:
+                isPrime = False
+                break
+        if isPrime:
+            max = i
```
Comment (dionyziz) [inline prob-3/prob3.py]: for..else

File: prob-4/prob4.py [hunk 45]
```diff
@@ -0,0 +1,13 @@
...
+        if isPalindrome(i * j):
+            if i * j > max:
+                max = i * j
+                break
+print max
```
Comment (dionyziz) [inline prob-4/prob4.py]: Use comprehension + `max` built-in function

File: prob-5/prob5.py [hunk 46]
```diff
@@ -0,0 +1,11 @@
+i = 20
+while True:
+    found = True
+    for j in range(1,20):
```
Comment (dionyziz) [inline prob-5/prob5.py]: Space after `,`

File: prob-5/prob5.py [hunk 47]
```diff
@@ -0,0 +1,11 @@
...
+        if i % j != 0:
+            found = False
+            break
+    if found:
```
Comment (dionyziz) [inline prob-5/prob5.py]: for..else

File: prob-10/prob10.py [hunk 48]
```diff
@@ -0,0 +1,12 @@
+def is_prime(num):
+    """Checks if the given number is a prime"""
+    is_prime = True
+    for i in range(2, num // 2 + 1):
+        if num % i == 0:
+            is_prime = False
+            break
```
Comment (dionyziz) [inline prob-10/prob10.py]: `return False`

File: prob-10/prob10.py [hunk 49]
```diff
@@ -0,0 +1,12 @@
...
+    return is_prime
```
Comment (dionyziz) [inline prob-10/prob10.py]: `return True`

File: prob-10/prob10.py [hunk 50]
```diff
@@ -0,0 +1,12 @@
+def is_prime(num):
+    """Checks if the given number is a prime"""
+    is_prime = True
```
Comment (dionyziz) [inline prob-10/prob10.py]: no need for this flag

File: prob-9/prob9.py (hunk 42 above)
Comment (vitsalis) [inline prob-9/prob9.py:12]: I think for loops are clearer in this case

File: prob-10/prob10.py [hunk 51]
```diff
@@ -0,0 +1,12 @@
...
+    for i in range(2, num // 2 + 1):
+        if num % i == 0:
+            is_prime = False
+            break
+    return is_prime
+
+primes = [x for x in range(2, 2 * 10 ** 6) if is_prime(x)]
```
Comment (dionyziz) [inline prob-10/prob10.py]: Use `filter`

File: prob-3/prob3.py [hunk 52]
```diff
@@ -0,0 +1,12 @@
+max = 1
+num = 600851475143
+
+for i in range(3, 10 ** 7):
```
Comment (dionyziz) [inline prob-3/prob3.py]: If you reverse the range, you can exit early

File: prob-4/prob4.py [hunk 53]
```diff
@@ -0,0 +1,13 @@
...
+    num = str(num)
+    return num == num[::-1]
+
+max = 1
+for i in range(100, 999):
+    for j in range(100, 999):
+        if is_palindrome(i * j):
+            if i * j > max:
+                max = i * j
+                break
```
Comment (dionyziz) [inline prob-4/prob4.py]: Collect all palindromes, then extract the maximum using `max`

File: prob-7/prob7.py [hunk 54]
```diff
@@ -0,0 +1,17 @@
+def is_prime(num):
+    """Checks if the number is a prime"""
+    is_prime = True
+    for i in range(2, num - 1):
+        if num % i == 0:
+            is_prime = False
+            break
+    return is_prime
```
Comment (dionyziz) [inline prob-7/prob7.py:8]: Split this function into a library, import it, and reuse.

File: prob-9/prob9.py (hunk 42 above)
Comment (dionyziz) [inline prob-9/prob9.py:12]: Agreed. "Flat is better than nested".

**Pull request merged**